ios/RNSScreenController.mm

```diff
@@ -1,73 +1,201 @@
+#if RN_FABRIC_ENABLED
 #import <react/renderer/components/rnscreens/RCTComponentViewHelpers.h>
+#else
+#import "RNSScreen.h"
+#import "RNSScreenStack.h"
+#import "RNSScreenStackHeaderConfig.h"
+#endif
+
 #import "RNSScreenComponentView.h"
 #import "RNSScreenWindowTraits.h"
 
 @implementation RNSScreenController {
+#if RN_FABRIC_ENABLED
   RNSScreenComponentView *_initialView;
+#else
+  __weak id _previousFirstResponder;
+  CGRect _lastViewFrame;
+  UIView *_fakeView;
+  CADisplayLink *_animationTimer;
+  CGFloat _currentAlpha;
+  BOOL _closing;
+  BOOL _goingForward;
+  int _dismissCount;
+  BOOL _isSwiping;
+  BOOL _shouldNotify;
+#endif
 }
 
+#pragma mark-- Common
 - (instancetype)initWithView:(UIView *)view
 {
   if (self = [super init]) {
     self.view = view;
+#if RN_FABRIC_ENABLED
     if ([view isKindOfClass:[RNSScreenComponentView class]]) {
       _initialView = (RNSScreenComponentView *)view;
     } else {
       RCTLogError(@"ScreenController can only be initialized with ScreenComponentView");
     }
+#else
+    _shouldNotify = YES;
+    _fakeView = [UIView new];
+#endif
   }
   return self;
 }
 
-- (void)setViewToSnapshot:(UIView *)snapshot
-{
-  [self.view removeFromSuperview];
-  self.view = snapshot;
-}
-
-- (void)resetViewToScreen
-{
-  [self.view removeFromSuperview];
-  self.view = _initialView;
-}
-
 // TODO: Find out why this is executed when screen is going out
 - (void)viewWillAppear:(BOOL)animated
 {
   [super viewWillAppear:animated];
+#if RN_FABRIC_ENABLED
   [RNSScreenWindowTraits updateWindowTraits];
   [_initialView notifyWillAppear];
+#else
+  if (!_isSwiping) {
+    [((RNSScreenView *)self.view) notifyWillAppear];
```

Comment:
I think all of this should be added on both archs, I would only exclude `[self hideHeaderIfNecessary];` to the moment we add `searchBar` on Fabric. Also, if there is no code with progress notification yet, we should exclude it on Fabric too.